import AppError from '@shared/errors/AppError'
import Faker from 'faker'

import Fake{{pascalCase moduleName}}Repository from '@modules/{{camelCase moduleName}}/repositories/fakes/Fake{{pascalCase moduleName}}Repository'
import {{pascalCase useCaseName}}Service from './{{pascalCase useCaseName}}.service'

let fake{{camelCase moduleName}}Repository: Fake{{pascalCase moduleName}}Repository
let {{camelCase useCaseName}}Service: {{pascalCase useCaseName}}Service

describe('{{pascalCase moduleName}} {{pascalCase useCaseName}}', () => {
  beforeEach(() => {
    fake{{camelCase moduleName}}Repository = new Fake{{pascalCase moduleName}}Repository()
    {{camelCase useCaseName}}Service = new {{pascalCase useCaseName}}Service(fake{{camelCase moduleName}}Repository)
  })

  it('should be able to XXXXXXXXXXXXX', async () => {
    const {{camelCase moduleName}} = await {{camelCase useCaseName}}Service.execute()

    expect({{camelCase moduleName}}).toHaveProperty('XXXXXXXXXXXXX')
  })

  it('should not be able to XXXXXXXXXXXXX', async () => {
    await expect(
      {{camelCase useCaseName}}Service.execute()
    ).rejects.toBeInstanceOf(AppError)
  })
})
